packages/rocketchat-ldap/server/ldap.js

```diff
@@ -191,6 +191,27 @@ export default class LDAP {
 
 		const usernameFilter = this.options.User_Search_Field.split(',').map((item) => `(${ item }=${ username })`);
 
+		const query = {
+			username,
+			ldap: true,
+			type: 'user',
+		};
+		const user = RocketChat.models.Users.findOne(query);
+		if (typeof user !== 'undefined' &&
+			typeof user.services !== 'undefined' &&
+			typeof user.services.ldap !== 'undefined' &&
+			typeof user.services.ldap.id !== 'undefined' &&
+			typeof user.services.ldap.idAttribute !== 'undefined') {
+			let idFilter = new this.ldapjs.filters.EqualityFilter({
+				attribute: user.services.ldap.idAttribute,
+				value: new Buffer(user.services.ldap.id, 'hex'),
+			});
+			if (filter.length === 1) {
+				idFilter = new this.ldapjs.AndFilter({ filters: [idFilter, this.ldapjs.parseFilter(filter[0])] });
+			}
+			return idFilter;
```

Comment:
The logic for handling idFilter could be made more modular. For example, separating the ID filter creation into a helper function would improve readability and maintainability of the code.